atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/SweepPriorityCalculator.java

```diff
@@ -15,105 +15,157 @@
+                SafeArg.of("unsafeTables", safeAndUnsafeTableReferences.unsafeTableRefs()));
     }
 
-    private double getSweepPriority(SweepPriority oldPriority, SweepPriority newPriority) {
+    private double getSweepPriorityScore(SweepPriority oldPriority, SweepPriority newPriority) {
         if (AtlasDbConstants.hiddenTables.contains(newPriority.tableRef())) {
-            // Never sweep hidden tables
+            // Never sweep hidden tables.
             return 0.0;
         }
         if (!newPriority.lastSweepTimeMillis().isPresent()) {
-            // Highest priority if we've never swept it before
+            // Highest priority if we've never swept it before.
             return Double.MAX_VALUE;
         }
         if (oldPriority.writeCount() > newPriority.writeCount()) {
             // We just swept this, or it got truncated.
             return 0.0;
         }
+
+        if (StreamTableType.isStreamStoreValueTable(newPriority.tableRef())) {
+            return getStreamStorePriorityScore(newPriority);
+        } else {
+            return getNonStreamStorePriorityScore(oldPriority, newPriority);
+        }
+    }
+
+    private double getStreamStorePriorityScore(SweepPriority newPriority) {
+        long millisSinceSweep = System.currentTimeMillis() - newPriority.lastSweepTimeMillis().getAsLong();
+
+        if (millisSinceSweep < WAIT_BEFORE_SWEEPING_STREAM_STORE_VALUE_TABLE.toMillis()) {
+            return 0L;
+        }
+        if (newPriority.writeCount() > STREAM_STORE_VALUES_TO_SWEEP) {
+            return Double.MAX_VALUE;
+        }
+
+        // Since almost every stream store value is 1MB long, give each cell a weight of 100.
+        // It should grow up to just 100 * STREAM_STORE_VALUES_TO_SWEEP = 100k anyway due to the above check.
+        return 100 * newPriority.writeCount();
+    }
+
+    private double getNonStreamStorePriorityScore(SweepPriority oldPriority, SweepPriority newPriority) {
         long staleValuesDeleted = Math.max(1, oldPriority.staleValuesDeleted());
         long cellTsPairsExamined = Math.max(1, oldPriority.cellTsPairsExamined());
-        long writeCount = Math.max(1, oldPriority.writeCount());
+        long writeCount = Math.max(1, oldPriority.writeCount()); // TODO(tboam): bug? should this be newPriority?
         double previousEfficacy = 1.0 * staleValuesDeleted / cellTsPairsExamined;
         double estimatedCellTsPairsToSweep = previousEfficacy * writeCount;
         long millisSinceSweep = System.currentTimeMillis() - newPriority.lastSweepTimeMillis().getAsLong();
 
         long daysSinceLastSweep = TimeUnit.DAYS.convert(millisSinceSweep, TimeUnit.MILLISECONDS);
         if (writeCount <= 100 + cellTsPairsExamined / 100 && daysSinceLastSweep < 180) {
             // Not worth the effort if fewer than 1% of cells are new and we've swept in the last 6 months.
+            // TODO(tboam): bug? is this really 1%?
```

Comment:
Does it go below 1% after some point?  Does it tend to 1% or just pass through it at some point?

If it tends to 1% I'll keep it in because exceeding 1% isn't that important on very small tables.  But if it is never really 1% except by coincidence then I think this is a bug.

Reply:
OK, idea seems reasonable but just not clearly communicated.  Pushing a commit which pulls this out into a function and explains intent better.

I'm going to leave in the other todo as I still want to think about how writeCount compares between oldPriority and newPriority but that can come later.  Not changing behaviour now.